src/Controller/StoreApi/OrderApi/OrderApiController.php

```diff
@@ -188,7 +188,9 @@ public function giftcardStateData(SalesChannelContext $context, Request $request
      */
     public function deleteGiftCardStateData(SalesChannelContext $context, Request $request): JsonResponse
     {
-        $this->paymentStateDataService->deletePaymentStateDataFromContextToken($context->getToken());
+        $stateData = json_decode($request->request->get('stateData', ''), true);
+        $Id = $stateData['id'];
+        $this->paymentStateDataService->deletePaymentStateData($Id);
 
         return new JsonResponse(['token' => $context->getToken()]);
```

Comment:
Better to handle situation when `stateData` is not provided

```suggestion
        $stateData = json_decode($request->request->get('stateData', '[]'), true);
        if (empty($stateData)) {
           // @TODO handle this situation
        }
            
        $stateData = json_decode($request->request->get('stateData', ''), true);
        $Id = $stateData['id'];
        $this->paymentStateDataService->deletePaymentStateData($Id);

        return new JsonResponse(['token' => $context->getToken()]);
```